subprojects/gradle/cthulhu/src/main/kotlin/com/avito/plugin/MarkReportAsSourceAction.kt

```diff
@@ -0,0 +1,61 @@
+package com.avito.plugin
+
+import com.avito.logger.Logger
+import com.avito.report.ReportsApi
+import com.avito.report.model.GetReportResult
+import com.avito.report.model.ReportCoordinates
+import com.avito.time.TimeProvider
+import com.github.salomonbrys.kotson.jsonObject
+
+/**
+ * TMS will use reports marked with special analyzer_key as source of truth
+ *
+ * related tasks:
+ * CTHL-495
+ * MBS-6483
+ */
+class MarkReportAsSourceAction(
+    private val reportsApi: ReportsApi,
+    private val timeProvider: TimeProvider,
+    private val logger: Logger
+) {
+
+    fun mark(reportCoordinates: ReportCoordinates) {
+        val testSuiteVersion = timeProvider.nowInMillis()
+
+        logger.info("This is a new version [$testSuiteVersion] of full test suite for tms")
+
+        val reportId = tryGetId(reportCoordinates)
+
+        if (reportId != null) {
+            reportsApi.pushPreparedData(
+                reportId = reportId,
+                analyzerKey = "test_suite",
+                preparedData = jsonObject(
+                    "full" to true,
+                    "version" to testSuiteVersion
+                )
+            ).onFailure { error ->
+                logger.critical("Can't push prepared data: testSuite info", error)
+            }
+
+            reportsApi.setFinished(reportCoordinates).onFailure { error ->
+                logger.critical("Can't finish report for coordinates: $reportCoordinates", error)
```

Comment:
If it is critical do we need to fail the task? `warn` level will be enough

Reply:
it is critical for sure, we should react on this errors because it is broken functionality on our side
but failing whole run - no

Reply:
How will I found that I had an error there If the build was `green`?
If one of the reasons is going to TMS I consider failing a build without one of the key functionality